internal/handler/handler.go

```diff
@@ -0,0 +1,104 @@
+package handler
+
+import (
+	"io"
+	"log"
+	"mime"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/spitfy/urlshortener/internal/config"
+	"github.com/spitfy/urlshortener/internal/service"
+)
+
+type Handler struct {
+	service ServiceShortener
+}
+
+type ServiceShortener interface {
+	Add(link string) (string, error)
+	Get(hash string) (string, error)
+}
+
+func (h *Handler) Get(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		w.Header().Set("Allow", http.MethodGet)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	hash := r.PathValue("hash")
+	if len(hash) == 0 || len(hash) > service.CharCnt {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	link, err := h.service.Get(hash)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(err.Error()))
+		return
+	}
+
+	w.Header().Add("Location", link)
+	w.WriteHeader(http.StatusTemporaryRedirect)
+}
+
+func (h *Handler) Post(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		w.Header().Set("Allow", http.MethodPost)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	mediaType, _, err := mime.ParseMediaType(r.Header.Get("Content-Type"))
+	if err != nil || mediaType != "text/plain" {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
+
+	if err != nil || len(body) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	url, err := h.service.Add(string(body))
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		log.Fatal(err)
```

Comment:
Использование `log.Fatal` уронит сервер, а это абсолютно точно не то, что ожидается от веб сервиса 
(https://t.me/bushigo/10). В принципе, использование настолько жесткого прерывания уместно только при инициализации, ДО входа в основной процесс. Если сервер уже запустился, то любая ошибка, возникающая на любом уровне, должна подниматься до самого верха в списке вызовов и там уже обрабатываться. В данном случае - в виде выдачи пользователю кода BadRequest и записи в лог веб-сервера.
```suggestion
		return
```